python/sglang/srt/layers/rotary_embedding.py

```diff
@@ -684,7 +686,16 @@ def __init__(
             / yarn_get_mscale(self.scaling_factor, float(mscale_all_dim))
             * attn_factor
         )
-        self.device = device
+        if device is not None:
+            self.device = device
+        elif _is_xpu:
+            self.device = "xpu"
+        elif _is_npu:
+            self.device = "npu"
+        elif _is_cuda:
+            self.device = "cuda"
+        else:
+            self.device = "cpu"
```

Comment:
nit: we can have a util maybe named as "get_default_device()", and put it in the correct place.

doing if-else inside the function is too tedious.